evap/results/views.py

```diff
@@ -27,24 +32,55 @@ def delete_template_cache(evaluation):
 
 
 def _delete_template_cache_impl(evaluation):
+    _delete_evaluation_template_cache_impl(evaluation)
+    if evaluation.course.evaluations.count() > 1:
+        _delete_course_template_cache_impl(evaluation.course)
+
+
+def _delete_evaluation_template_cache_impl(evaluation):
     caches['results'].delete(get_evaluation_result_template_fragment_cache_key(evaluation.id, 'en', True))
     caches['results'].delete(get_evaluation_result_template_fragment_cache_key(evaluation.id, 'en', False))
     caches['results'].delete(get_evaluation_result_template_fragment_cache_key(evaluation.id, 'de', True))
     caches['results'].delete(get_evaluation_result_template_fragment_cache_key(evaluation.id, 'de', False))
 
 
+def _delete_course_template_cache_impl(course):
+    caches['results'].delete(get_course_result_template_fragment_cache_key(course.id, 'en'))
+    caches['results'].delete(get_course_result_template_fragment_cache_key(course.id, 'de'))
+
+
 def warm_up_template_cache(evaluations):
     evaluations = get_evaluations_with_prefetched_data(evaluations)
     current_language = translation.get_language()
+    courses_to_render = set([evaluation.course for evaluation in evaluations if evaluation.course.evaluations.count() > 1])
+    course_avg_grades = {}
+    course_weight_sums = {}
     try:
+        for course in courses_to_render:
+            course.distribution = calculate_average_course_distribution(course)
+            course.avg_grade = distribution_to_grade(course.distribution)
+            course_avg_grades[course.pk] = course.avg_grade
+            course_weight_sums[course.pk] = sum(evaluation.weight for evaluation in course.evaluations.all())
+            if any(evaluation.state != "published" for evaluation in course.evaluations.all()):
```

Comment:
If you're repeatedly using "course.evaluations.all()", this is generating a new QuerySet each time, so the query is not cached, right? I guess we want the caching here?